archunit/src/main/java/com/tngtech/archunit/core/importer/DomainBuilders.java

```diff
@@ -142,6 +143,11 @@ SELF withModifiers(Set<JavaModifier> modifiers) {
             return self();
         }
 
+        SELF withLineNumber(int lineNumber) {
```

Comment:
Should we name this method in a way that reflects that it does more than just stupidly adding properties like other `with`-methods? Because it does not seem to be the way of least surprise to me, that I might set `withLineNumber(9)`, but afterwards `lineNumber == 7` because I actually ignored that element? I don't know what the best name would be, maybe something like `recordMethodBodyLineNumber(..)`? Or `addMethodBodyLineNumber(..)`? (but that one suggests to me I would keep them all somehow) Or `registerMBLN`? Or `storeMinimumMethod{Body}LineNumber(..)`?

Reply:
Hehehe, I actually started with the `JavaMemberBuilder`'s field called `lineNumber` and had the following JavaDoc for `withLineNumber`:
```
/**
 * records a visited line number.
 * The actual line number of the JavaMember being build will be the smallest lineNumber != 0 ever visited.
 */
```
But then I thought that `withLineNumber` might not need that explanation if the field is called differently, i.e. `firstLineNumber`. 

If we need to keep all line numbers for some reason (e.g. add them to a `Set<Integer> lineNumbers`), we'd only need to change the implementation, but could keep the interface of this method.

I wouldn't include `Method`(`Body`) in the name, as the builder can be used for all members and is actually used for all code units. (It's not the builder's fault that we don't use call it for fields... 😜)

**TL;DR**: I think `recordLineNumber` would be the best alternative, but I'd still think that `withLineNumber` fits better for a builder.

Reply:
I'll buy `recordLineNumber` then :wink: For me that "`withLineNumber` fits for a builder" makes it even worse :rofl: Exactly because it looks like a typical builder method that you can quickly skim over but it actually does not behave like it. I'll rename that and merge it!